src/snake.cpp

```diff
@@ -417,19 +417,31 @@ void snake::snake::getInput()
             {
                 case KEY_UP:
                 case 'w':
-                    input = moveUp;
+                    if (!snake::snake::noOppDir)
+                        input = moveUp;
+                    if (input != moveDown)
```

Comment:
lol just noticede, that this doesnt make any sense. just do if (!noOppDir || noOppdir && input != moveDown). And noOppDir is a really bad name, probably change that into something diffrent, like "oppositeDirIllegal" or something.

Reply:
You can't check in an if statement if a boolean is true. That's why I wrote it that way.

Reply:
so first, an if statement works with boolean logic so that will work and seconds that implementation hassome problems, like you can't go backwards if snakeLength = 0, plus a bug, if you go up and press left+ down you actually go down, so it probably would be better to move the check into the actuall movement, or into a new function, that gets called by update() between input() and the snake movement function.

Reply:
i know these problems but this branch is just for the on/off switch and not bugfixing, because this problem has existed before. And you can try it yourself, you can't check if a bool is true.

Reply:
And your var name doesn't makes clear what its about -> as well as using acronyms is most of the time bad practise. Longer names are better than short ones sometimes.